File: src/main/java/com/estonianport/geservapp/controller/SalonController.java
package com.estonianport.geservapp.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;

import  com.estonianport.geservapp.commons.GeneralPath;
import com.estonianport.geservapp.model.Salon;
import com.estonianport.geservapp.service.SalonService;

@Controller
public class SalonController {

	@Autowired
	private SalonService salonService;

	@RequestMapping("/abmSalon")
	public String abm(Model model) {
		model.addAttribute("listaSalon", salonService.getAll());
		return GeneralPath.SALON + GeneralPath.PATH_SEPARATOR + GeneralPath.ABM_SALON;
	}

	@GetMapping("/saveSalon/{id}")
	public String showSave(@PathVariable("id") Long id, Model model) {
		if(id != null && id != 0) {
			model.addAttribute(GeneralPath.SALON, salonService.get(id));
		}else {
			model.addAttribute(GeneralPath.SALON, new Salon());
		}
		return GeneralPath.SALON + GeneralPath.PATH_SEPARATOR + GeneralPath.SAVE_SALON;
	}

	@PostMapping("/saveSalon")
	public String save(Salon salon, Model model) {
		salonService.save(salon);
		return GeneralPath.REDIRECT + GeneralPath.ABM_SALON;
	}

	@GetMapping("/deleteSalon/{id}")
	public String delete(@PathVariable("id") Long id, Model model) {
		salonService.delete(id);
		return GeneralPath.REDIRECT + GeneralPath.ABM_SALON;
	}
}

File: src/main/kotlin/com/estonianport/agendaza/controller/SalonController.kt
package com.estonianport.agendaza.controller

import com.estonianport.agendaza.model.Salon
import com.estonianport.agendaza.service.SalonService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
@CrossOrigin("*")
class SalonController {

    @Autowired
    lateinit var salonService: SalonService


    @GetMapping("/getAllSalon")
    fun abm(): MutableList<Salon>? {
        return salonService.getAll()
    }

    @GetMapping("/saveSalon/{id}")
    fun showSave(@PathVariable("id") id: Long): ResponseEntity<Salon>? {
        if (id != 0L) {
            return ResponseEntity<Salon>(salonService.get(id), HttpStatus.OK)
        }

        return ResponseEntity<Salon>(HttpStatus.NO_CONTENT)
    }

    @PostMapping("/saveSalon")
    fun save(@RequestBody salon: Salon): ResponseEntity<Salon> {
        return  ResponseEntity<Salon>(salonService.save(salon), HttpStatus.OK)
    }

    @DeleteMapping("/deleteSalon/{id}")
    fun delete(@PathVariable("id") id: Long): ResponseEntity<Salon> {

        //Elimina todos los eventos del salon 
        //val listaEventos: List<Evento> = eventoService.getEventosBySalon(salonService.get(id))
        //for (evento in listaEventos) {
          //  eventoService.delete(evento.getId())
        //}

        //Elmina el salon
        salonService.delete(id)
        return ResponseEntity<Salon>(HttpStatus.OK)
    }
}